drag_and_drop_v2/public/js/drag_and_drop.js

```diff
@@ -1453,8 +1474,42 @@ function DragAndDropBlock(runtime, element, configuration) {
                     cancelAnimationFrame(raf_id);
                 }
                 raf_id = requestAnimationFrame(function() {
-                    var dx = pageX(evt) - drag_origin.x;
-                    var dy = pageY(evt) - drag_origin.y;
+                    var x = pageX(evt);
+                    var y = pageY(evt);
+
+                    var $target = $container.find('.target');
+                    var $target_rect = $target.get(0).getBoundingClientRect();
+                    var top = $target_rect.top + window.scrollY;
+                    var bottom = $target_rect.bottom + window.scrollY;
+                    var left = $target_rect.left + window.scrollX - 50;
+                    var right = $target_rect.right + window.scrollX + 50;
+
+                    if (top <= y && bottom >= y && left <= x && right >= x) {
+                        var viewportTop = window.scrollY + fixedHeaderHeight;
+                        var viewportBottom = window.scrollY + $(window).height();
+                        var viewportLeft = window.scrollX;
+                        var viewportRight = window.scrollX + $(window).width();
+
+                        if (x < (viewportLeft + 50)) {
+                            $target.scrollLeft($target.scrollLeft() - 50);
+                        }
+
+                        if (x > (viewportRight - 50)) {
+                            $target.scrollLeft($target.scrollLeft() + 50);
+                        }
+
+                        if (y > (viewportBottom - 50)) {
+                            $document.scrollTop($document.scrollTop() + 50);
+                        }
+
+                        if (top < viewportTop + fixedHeaderHeight && y < (viewportTop + fixedHeaderHeight + 50)) {
```

Comment:
Since `fixedHeaderHeight` is already already included in `viewportTop` (line 1488), should we remove the two occurrences of `fixedHeaderHeight` here?

Reply:
Good catch. Fixed.